src/network/networkd-can.c

```diff
@@ -152,6 +152,22 @@ static int link_set_can(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_CAN_CTRLMODE attribute: %m");
         }
 
+        if (link->network->can_termination >= 0) {
+                uint64_t termination = link->network->can_termination;
+
+                if (termination > UINT16_MAX) {
+                        log_link_error(link, "invalid can-termination");
+                        return -ERANGE;
+                }
```

Comment:
Please do not check the value range here, but introduce, say `config_parse_can_termination()`, and check the value inside the function.